pycbc/bank/conversions.py

```diff
@@ -0,0 +1,114 @@
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+"""
+This module is supplied to make a convenience function for converting into
+specific values from PyCBC template banks.
+"""
+
+from pycbc import conversions as conv
+from pycbc import pnutils
+
+# Convert from parameter name to helper function
+# some multiple names are used for the same function
+conversion_options = ['mass1', 'mass2', 'spin1z', 'spin2z', 'duration',
+                      'template_duration', 'mtotal', 'total_mass',
+                      'q', 'invq', 'eta', 'chirp_mass', 'mchirp',
+                      'chieff', 'chi_eff', 'effective_spin', 'chi_a']
+
+
+mass_conversions = {
+    'mtotal': conv.mtotal_from_mass1_mass2,
+    'total_mass': conv.mtotal_from_mass1_mass2,
+    'q': conv.q_from_mass1_mass2,
+    'invq': conv.invq_from_mass1_mass2,
+    'eta': conv.eta_from_mass1_mass2,
+    'mchirp': conv.mchirp_from_mass1_mass2,
+    'chirp_mass': conv.mchirp_from_mass1_mass2,
+}
+
+spin_conversions = {
+    'chieff': conv.chi_eff,
+    'chi_eff': conv.chi_eff,
+    'effective_spin': conv.chi_eff,
+    'chi_a': conv.chi_a
+}
+
+
+def bank_conversion(parameter, bank, template_ids):
+    """ Get a specific value from a hdf file object in standard PyCBC
+    template bank format
+
+    Parameters
+    ----------
+    parameter: str
+        the parameter to convert to, must be in conversion_options
+
+    bank: h5py File object or dictionary of arrays
+        Template bank containing the parameters for use in conversions
+        must contain mass1, mass2, spin1z, spin2z as a minimum
+
+    template_ids: numpy array
+        Array of template IDs for reading a set of templates from the bank
+
+    Returns
+    -------
+    values: numpy array, same size as template_ids
+        Array of whatever the requested parameter is calculated for
+        the specified templates in the bank
+
+    """
+    # These just give things already in the bank
+    if parameter in bank:
+        values = bank[parameter][:][template_ids]
+
+    # These things may be in the bank, but if not, we need to calculate
+    elif parameter in ['template_duration', 'duration']:
+        if 'template_duration' in bank:
+            # This statement should be the reached only if 'duration'
+            # is given, but 'template_duration' is in the bank
+            values = bank['template_duration'][:][template_ids]
+        else:
+            values = pnutils.get_imr_duration(bank['mass1'][:][template_ids],
+                                              bank['mass2'][:][template_ids],
+                                              bank['spin1z'][:][template_ids],
+                                              bank['spin2z'][:][template_ids],
+                                              bank['f_lower'][:][template_ids],
+                                              approximant="SEOBNRv4")
```

Comment:
Not sure if SEOBNRv4 ought to be hard-coded here - in principle I think it should be allowed to be a kwarg, eg `def bank_conversion(parameter, bank, template_ids, duration_approximant='SEOBNRv4'):`